presto-hive/src/main/java/com/facebook/presto/hive/HiveClientConfig.java

```diff
@@ -1642,6 +1645,18 @@ public Duration getPartitionLeaseDuration()
         return partitionLeaseDuration;
     }
 
+    @Config("hive.materialized-view-missing-partition-predicate-count-threshold")
+    @ConfigDescription("Max number of partition predicates missing in the materialized view before the base table query string becomes too long")
+    public HiveClientConfig setMaterializedViewMissingPartitionPredicateCountThreshold(int materializedViewMissingPartitionPredicateCountThreshold)
+    {
+        this.materializedViewMissingPartitionPredicateCountThreshold = this.materializedViewMissingPartitionPredicateCountThreshold;
+        return this;
+    }
+
+    public int getMaterilizedViewMissingPartitionPredicateCountThreshold()
+    {
+        return this.materializedViewMissingPartitionPredicateCountThreshold;
+    }
```

Comment:
Please add a single blank line between methods

Reply:
@ConfigDescription("Max number of partitions allowed for missing predicates in the materialized view for query optimization.")

Reply:
Lets change the config name to
materialized-view-missing-partitions-threshold

Reply:
Also reflect the same in the config desciption
    @ConfigDescription("Max number of missing partitions allowed in the materialized view for query optimization")

Reply:
People who do not have context might find it hard to understand this description. Would be great to include the following points:
1. Is this applied at write time or query time?
2. What are the expected behaviors when it is below or above the threshold?

Reply:
Good point @yuanzhanhku . How about 
`Threshold for missing partition count, after which the partition is considered un-materialized, for read query optimization` ?
@jainxrohit

Reply:
How about:
Materialized views with missing partitions more than this threshold are disabled at read time and fall back to read the base tables.

Reply:
I would not use the work disabled, it can mean multiple things.
Materialized views with missing partitions more than this threshold falls back to the base tables at read time.

Reply:
I agree with @jainxrohit on this - "Materialized views with missing partitions more than this threshold falls back to the base tables at read time". 

Will change it to that.